sql/parser/eval.go

```diff
@@ -319,109 +319,236 @@ type cmpArgs struct {
 	rightType reflect.Type
 }
 
-var cmpOpResultType = reflect.New(reflect.TypeOf(cmpOps).Elem().Out(0)).Elem().Interface().(DBool)
+type cmpOp struct {
+	fn     func(Datum, Datum, cmpOpCacheVal) (DBool, error)
+	initFn func(Datum, Datum) (cmpOpCacheVal, error)
+}
+
+type cmpOpCacheVal interface{}
```

Comment:
This type is just unnecessary abstraction at this point. I'd just use `interface{}` directly.